pulsar-client/src/main/java/org/apache/pulsar/client/impl/auth/oauth2/protocol/TokenClient.java

```diff
@@ -73,10 +73,21 @@ public void close() throws Exception {
 
     /**
      * Constructing http request parameters.
-     * @param bodyMap List of parameters to be requested.
+     * @param req object with relevant request parameters
      * @return Generate the final request body from a map.
      */
-    String buildClientCredentialsBody(Map<String, String> bodyMap) {
+    String buildClientCredentialsBody(ClientCredentialsExchangeRequest req) {
+        Map<String, String> bodyMap = new TreeMap<>();
```

Comment:
Could you explain why do you use a `TreeMap` here? I'm not sure if the order makes a difference.

Reply:
It's only a `TreeMap` because I copied it from `exchangeClientCredentials`. I don't remember any specific reason for using this data structure.